smb/src/resource.rs

```diff
@@ -336,7 +350,7 @@ impl ResourceHandle {
             flags: QueryInfoFlags::new()
                 .with_restart_scan(true)
                 .with_return_single_entry(true),
-            file_id: self.file_id,
+            file_id: self.file_id()?,
```

Comment:
In these instances, if there were any thread races possible, the above atomic wouldn't protect you for the duration of this call. It could be `close`ed right after you got it. The only way to guard against that is some sort of async lock. However, you don't need that as you can't race.